eng/common/scripts/modules/Package-Properties/Package-Properties.psm1

```diff
@@ -0,0 +1,172 @@
+        $jsStylePkgName = $pkgName.replace("azure-", "@azure/")
+        if ($projectJson.name -eq "$jsStylePkgName")
+        {
+            $pkgReadMePath = Join-Path $pkgPath "README.md"
+            $pkgReadMePath = If (Test-Path ($pkgReadMePath)) {  $pkgReadMePath } Else { $null }
+            $pkgChangeLogPath = Join-Path $pkgPath "CHANGELOG.md"
+            $pkgChangeLogPath = If (Test-Path ($pkgChangeLogPath)) {  $pkgChangeLogPath } Else { $null }
+
+            return @{
+                pkgName = $projectJson.name;
+                pkgVersion = $projectJson.version;
+                pkgDirPath = $pkgPath;
+                pkgReadMePath = $pkgReadMePath;
+                pkgChangeLogPath = $pkgChangeLogPath;
+            }
+        }
+    }
+    return $null
+}
+
+function Extract-PythonPkgProps ($pkgPath, $pkgName)
+{
+    $pkgName = $pkgName.Replace('_', '-')
+
+    if (Test-Path (Join-Path $pkgPath "setup.py"))
+    {
+        $setupLocation = $pkgPath.Replace('\','/')
+        pushd $RepoRoot
+        $setupProps = (python -c "import scripts.devops_tasks.common_tasks; obj=scripts.devops_tasks.common_tasks.parse_setup('$setupLocation'); print('{0},{1}'.format(obj[0], obj[1]));") -split ","
+        Write-Host $setupProps
+        popd
+        if (($setupProps -ne $null) -and ($setupProps[0] -eq $pkgName))
+        {
+            $pkgReadMePath = Join-Path $pkgPath "README.md"
+            $pkgReadMePath = If (Test-Path ($pkgReadMePath)) {  $pkgReadMePath } Else { $null }
+            $pkgChangeLogPath = Join-Path $pkgPath "CHANGELOG.md"
+            $pkgChangeLogPath = If (Test-Path ($pkgChangeLogPath)) {  $pkgChangeLogPath } Else { $null }
+
+            return @{
+                pkgName = $setupProps[0];
+                pkgVersion = $setupProps[1];
+                pkgDirPath = $pkgPath;
+                pkgReadMePath = $pkgReadMePath;
+                pkgChangeLogPath = $pkgChangeLogPath;
+            }
+        }
+    }
+    return $null
+}
+
+function Extract-JavaPkgProps ($pkgPath, $pkgName)
+{
+    $projectPath = Join-Path $pkgPath "pom.xml"
+
+    if (Test-Path $projectPath)
+    {
+        $projectData = New-Object -TypeName XML
+        $projectData.load($projectPath)
+        $projectPkgName = Select-XML -Xml $projectData -XPath '/*[local-name()="project"]/*[local-name()="artifactId"]/text()'
+        $pkgVersion = Select-XML -Xml $projectData -XPath '/*[local-name()="project"]/*[local-name()="version"]/text()'
+
+        $pkgReadMePath = Join-Path $pkgPath "README.md"
+        $pkgReadMePath = If (Test-Path ($pkgReadMePath)) {  $pkgReadMePath } Else { $null }
+        $pkgChangeLogPath = Join-Path $pkgPath "CHANGELOG.md"
+        $pkgChangeLogPath = If (Test-Path ($pkgChangeLogPath)) {  $pkgChangeLogPath } Else { $null }
+        
+        if ($projectPkgName.ToString() -eq $pkgName)
+        {
+            return @{
+                pkgName = $pkgName;
```

Comment:
We should probably turn this into a class and return that class so that we have a consistent structure for everyone.  It also will allow us to wrap some of the common functionality like the README and CHANGELOG probing directly in the class so we don't have to duplicate that everywhere.